gwpy/timeseries/tests/test_timeseries.py

```diff
@@ -1160,6 +1160,34 @@ def test_resample_noop(self):
             new = data.resample(data.sample_rate)
             assert data is new
 
+    def test_upsample_repeat_factor_3(self, gw150914):
+        """Test that repeat upsampling by a factor of 3 works as intended.
+        """
+        factor = 3
+        upsamped = gw150914.upsample_repeat(factor)
+        upsamped.xindex  # generate the x index
+        assert upsamped.sample_rate == factor * gw150914.sample_rate
+        for valj, val in enumerate(gw150914):
+            for fj in range(factor):
+                # Each index of input valj maps to an index factor * valj
+                upj = factor * valj + fj  # index of each repeated sample
+                numpy.testing.assert_equal(val, upsamped[upj])
+
+    def test_upsample_repeat_factor_3_xindex(self, gw150914):
+        """Test that a repeat upsampled (3X) timeseries has a correct xindex.
+        """
+        factor = 3
+        upsamped = gw150914.upsample_repeat(factor)
+        upsamped.xindex  # generate the x index
+        assert len(upsamped) == len(upsamped.xindex)
+
+    def test_upsample_repeat_float_exception(self, gw150914):
+        """Assert that upsampling by a non-integer factor raises a ValueError.
+        """
+        factor = 3.3
+        with pytest.raises(ValueError):
+            gw150914.upsample_repeat(factor)
```

Comment:
Can this be simplified to not have a variable assignment?

```suggestion
        with pytest.raises(ValueError):
            gw150914.upsample_repeat(3.3)
```